### PR TITLE
[dist, config, docs] feat: drop FSDP1

### PR DESCRIPTION
**Pull request by Luosuu**
## Summary

FSDP1 has been deprecated in PyTorch and in VeOmni for some time
(``.agents/knowledge/`` already labelled it legacy). This PR removes
every FSDP1 path in VeOmni so the codebase ships only FSDP2 + DDP.

- Delete ``veomni/distributed/fsdp/`` and the ``parallelize_model_fsdp1``
  dispatcher branch in ``torch_parallelize.py``. ``build_parallelize_model``
  loses the FSDP1-only kwargs (``enable_full_shard``,
  ``enable_shard_grad_op``, ``use_orig_params``,
  ``enable_fsdp_offload``); the dispatcher now resolves to FSDP2 or DDP.
- Narrow ``ParallelState.dp_mode`` Literal to ``{"ddp", "fsdp2"}`` and
  bump its default to ``"fsdp2"``. Drop FSDP1-only knobs from
  ``FSDPConfig`` (``full_shard``, ``offload``) and
  ``MixedPrecisionConfig`` (``buffer_dtype``).
- Add an ``FSDPConfig.__post_init__`` validator that raises with a
  migration hint when a stale config still passes ``fsdp_mode="fsdp1"``
  (dataclass ``Literal`` is not enforced at runtime).
- Migrate every YAML under ``configs/`` from ``fsdp_mode: fsdp1`` to
  ``fsdp_mode: fsdp2`` + ``init_device: meta`` (FSDP2 requires meta
  init). Strip leftover ``full_shard`` / ``fsdp_config.offload`` from
  the remaining configs.
- Refresh checkpoint / optim / model / LoRA helpers to drop FSDP1
  bookkeeping (DCP backend allowlist, ``_fsdp_wrapped_module`` prefix
  normalization, ``load_peft_shard_states``, stale comments).
- Update every doc that referenced FSDP1 (``arguments.md``,
  ``basic_modules.md``, ``ep_fsdp2.md``, ``lora.md``,
  ``extra_parallel.md``, ``ulysses.md``, ``dit_model_guide.md``, NPU
  support matrix) and the ``.agents/knowledge`` / ``veomni-develop``
  skill.
- Drop the now-unused ``enable_full_shard`` / ``enable_fsdp_offload``
  kwargs from ``tests/utils/test_extra_parallel_clip_grad_norm.py`` and
  ``tests/optim/test_muon_fsdp2_smoke.py`` (both already run with
  ``fsdp_mode=fsdp2``).

Net diff: 49 files changed, ~1500 lines removed.

## Test plan

- [x] ``ruff check`` and ``ruff format --check`` clean on every touched
      file (the pre-existing ``tests/models/test_models_logits_equal_v5.py``
      ruff failure on ``main`` is unrelated).
- [x] Every migrated YAML round-trips through ``FSDPConfig`` and
      asserts ``fsdp_mode == "fsdp2"``.
- [x] ``FSDPConfig(fsdp_mode="fsdp1")`` raises ``ValueError`` with a
      migration hint.
- [ ] CI: ``pytest tests/parallel/``, ``tests/checkpoints/``,
      ``tests/optim/test_muon_fsdp2_*``, and ``tests/utils/
      test_extra_parallel_clip_grad_norm.py`` on multi-GPU CI (these
      need a real distributed runtime, not exercised locally).
- [ ] Optional smoke-run of one migrated config (e.g.
      ``configs/text/qwen3.yaml``) on a GPU host.

## Backwards compatibility

Hard break: stale configs that still set ``fsdp_mode: fsdp1`` will fail
at parse time with the new validator error pointing users at
``fsdp_mode: fsdp2`` + ``init_device: meta``. This is intentional per the
design discussion — there is no silent compat shim.

### CODE REVIEW

**Review by TimYangst — APPROVED** [review]
Thanks for the cleanup!

File: veomni/arguments/arguments_types.py [hunk 1]
```diff
@@ -293,18 +288,10 @@ class FSDPConfig:
...
     )
-    full_shard: bool = field(
-        default=True,
-        metadata={"help": "Enable fully shard for FSDP training (ZeRO-3)."},
-    )
     forward_prefetch: bool = field(
         default=True,
         metadata={"help": "Enable forward prefetch."},
     )
-    offload: bool = field(
```
Comment (Coach257) [inline veomni/arguments/arguments_types.py:304]: Maybe keep `offload` as we got #753

**Review by Coach257 — APPROVED** [review]
